django/db/models/expressions.py

```diff
@@ -286,8 +286,14 @@ def _resolve_output_field(self):
         """
         sources_iter = (source for source in self.get_source_fields() if source is not None)
         for output_field in sources_iter:
+            source = None
             if any(not isinstance(output_field, source.__class__) for source in sources_iter):
-                raise FieldError('Expression contains mixed types. You must set output_field.')
+                raise FieldError('Expression contains mixed types: '
```

Comment:
Please restore the second sentence `You must set output_field.`. It should remain untouched.

Reply:
Will do.